code/modules/holidays/halloween/decorators.dm

```diff
@@ -0,0 +1,98 @@
+/datum/game_decorator/halloween
+
+/datum/game_decorator/halloween/is_active_decor()
+	return (get_event_progress() != -1)
+
+/datum/game_decorator/halloween/proc/get_event_progress()
+	. = -1
+	var/cur_day = text2num(time2text(world.timeofday, "DD"))
+	var/cur_mon = text2num(time2text(world.timeofday, "MM"))
+	if(cur_mon == 10)
+		return min(0, 31 - cur_day)
```

Comment:
max is probably what you intended? But even then if this returned a scalar 0-1 on how haloweeny it is it probably would make more sense than -1, 0-31.

So I think you could `return 1 - (31-cur_day)/31`? to get a value 0-1 where the first is 0.032258064516129 and the 31st is 1?